### ISSUE TITLE
Please implement Container Networking Interface (CNI)

### ISSUE THREAD

**Issue by AceHack** [Azure/service-fabric-issues#726]
Most container orchestrators have started using CNI, please follow suit.
https://github.com/containernetworking/cni

Calico would be a great provider of CNI to use as a first class plugin
https://www.projectcalico.org/

**Comment by RajeetN** [comment]
Yes this is in the works. Is this an on premise (standalone) deployment you need this for?

**Comment by AceHack** [comment]
Why is this closed if it's in the works?  Is there some way to track it, should this be labeled roadmap?  I don't understand why this was closed.  It's for both, some deployments are on-premise and some are in Azure.

**Comment by RajeetN** [comment]
Could you elaborate on the scenario where you need it to be CNI on Azure?

**Comment by AceHack** [comment]
Yes, looking to allow for automation of infrastructure native as part of app development, using native app development tools to express networking rules.  Auto setting up LB rules, multiple different ingresses with different routing rules, network policy, service mesh, automatic iptables like routing of services with immutable IPs, automatic tracing of communications, network namespaces that can be shared by multiple containers i.e. pods, immutable infrastructure, software-defined networking (SDN), software load balancers inside the mesh that don't need any Azure or external load balancers, consistency between on-premises and Azure environment, software-defined security, cheap, external resource discovery, egress, host protection, ability to use standard tools like wireshare for debugging container network traffic, the list goes on and on.

**Comment by AceHack** [comment]
Also, I would just like to say I don't care if in Auze there is an azure specific CNI plugin used and on-premises I use a calico based CNI plugin, I just want to have a consistent application development, deployment, and management experience no matter where I run my portable application.  The Azure-based CNI plugin should not overuse cloud infrastructure and make it more expensive than needed, wherever a in cluster solution makes sense it should be used instead like in cluster load balancing.  Basically I want Layer 2-7 SDN built in Service Fabric that only call out to external out of cluster devices where absolutly necessary but all of that should be automated though a consistent development experience.

**Comment by AceHack** [comment]
Oh I hope it all goes without saying but all of this needs to support IPv6 not just IPv4 since we are in the IoT age.  :)
